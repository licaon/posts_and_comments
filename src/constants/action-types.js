export const INIT = 'INIT';

export const START_LOADING_DATA = 'START_LOADING_DATA';
export const END_LOADING_DATA = 'END_LOADING_DATA';

export const GET_POSTS = 'GET_POSTS';
export const ADD_POST = 'ADD_POST';
export const DELETE_POST = 'DELETE_POST';

export const ADD_SEARCH_FILTER = 'ADD_SEARCH_FILTER';
export const ADD_SELECT_FILTER = 'ADD_SELECT_FILTER';

export const GET_USERS = 'GET_USERS';
export const ADD_USER = 'ADD_USER';
