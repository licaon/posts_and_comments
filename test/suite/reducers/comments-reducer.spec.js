import { ADD_COMMENTS, DELETE_POST } from 'src/constants/action-types';
import CommentsReducer from 'src/reducers/comments-reducer';

describe('CommentsReducer', () => {
  it(`should update reducer correctly on ${ADD_COMMENTS}`, (done) => {
    expect(CommentsReducer(
      undefined,
      {
        type: ADD_COMMENTS,
        payload: {
          postId: 1,
          comments: [2],
        },
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should update reducer correctly on ${DELETE_POST}`, (done) => {
    expect(CommentsReducer(
      {
        2: 'to delete',
      },
      {
        type: DELETE_POST,
        payload: {
          id: 2,
        },
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should return reducer correctly on unknown action`, (done) => {
    expect(CommentsReducer(
      undefined,
      { type: 'unknown' },
    )).toMatchSnapshot();
    done();
  });
});
