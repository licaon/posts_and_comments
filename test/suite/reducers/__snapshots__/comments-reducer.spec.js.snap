// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsReducer should return reducer correctly on unknown action 1`] = `Object {}`;

exports[`CommentsReducer should update reducer correctly on ADD_COMMENTS 1`] = `
Object {
  "1": Array [
    2,
  ],
}
`;

exports[`CommentsReducer should update reducer correctly on DELETE_POST 1`] = `Object {}`;
